### ISSUE TITLE
State group.present fails to create a new group on Windows

### ISSUE THREAD

**Issue by mts-avco** [saltstack/salt#45503]
### Description of Issue/Question
Asserting a group under Windows fails if the group does not exist.

Manually altering `salt\states\group.py:249` directly to only pass 1 argument then causes the state to apply and the group to be created.

### Setup
Install a salt-minion on Windows and authorise key as usual.

top.sls:
```
base:
  '*':
    group1
```

group1.sls:
```
newgroup1:
  group.present: []
```

### Steps to Reproduce Issue
The error reported on the master from `salt win-test state.apply` is as follows:
```
          ID: newgroup1
    Function: group.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "c:\salt\bin\lib\site-packages\salt\state.py", line 1843, in call
                  **cdata['kwargs'])
                File "c:\salt\bin\lib\site-packages\salt\loader.py", line 1795, in wrapper
                  return f(*args, **kwargs)
                File "c:\salt\bin\lib\site-packages\salt\states\group.py", line 249, in present
                  system=system):
              TypeError: add() takes exactly 1 argument (3 given)
     Started: 15:01:29.693000
    Duration: 110.0 ms
     Changes:
```

### Versions Report
```
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: 1.10.0
       cherrypy: 10.2.1
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.3
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)]
   python-gnupg: 0.4.0
         PyYAML: 3.11
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.3
        timelib: 0.2.4
        Tornado: 4.5.1
            ZMQ: 4.1.6

System Versions:
           dist:
         locale: cp1252
        machine: AMD64
        release: 2016Server
         system: Windows
        version: 2016Server 10.0.14393  Multiprocessor Free
```

**Comment by Ch3LL** [comment]
Should be resolved by https://github.com/saltstack/salt/pull/44080

can you give that a try?

**Comment by mts-avco** [comment]
Just to let you know I haven't forgotten about this, I have the updated files but haven't yet put them on a machine to test. I hope to do so and close this some time this week.

**Comment by mts-avco** [comment]
I tried, but I'm unable to apply the updated files as suggested by @twangboy, most likely due to the `import salt.utils.win_functions` modification suggested by @gtmanfred. If someone can let me know what changes are required, or supply changed files I can simply drop in, then I'll give it another go.

**Comment by mts-avco** [comment]
Using the files without any changes gives the following error:

```
local:
----------
          ID: WebAdmin
    Function: group.present
      Result: False
     Comment: State 'group.present' was not found in SLS 'test'
              Reason: 'group' __virtual__ returned False: No module named platform
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   0.000 ms
```

**Comment by twangboy** [comment]
There are 2 things you could try here:
```
newgroup1:
  group.present
```
or
```
create_group:
  group.present:
    - name: newgroup1
```

**Comment by twangboy** [comment]
I can confirm that this is fixed in 2017.7.3
